### DISCUSSION TITLE
One fragment (?) replacing another (or cross-fading)

### DISCUSSION THREAD

**Discussion by Sciss** [hakimel/reveal.js#3299] [Q&A] [answered]
I want to transition between one text fragment (ideally `span`) and another text fragment within the same slide. So I guess I need something like fragment. But the problem is the first fragment doesn't change `display` to `none`, so it still occupies space, and the second fragment is shown vertically below the first fragment, rather than taking the exact same location.

```html
They said <span class="fragment fade-out">monday</span><span class="fragment fade-in">mundane</span>.
```
The intention is that the slide opens with __They said monday__, and then the 'monday' disappears and is replaced by 'mundane', so it reads __They said mundane.__ I cannot use two separate slides, because there is a video playing in the same slide.

**Comment by Sciss** [comment, marked as answer]
After more searching, I found [this issue](https://github.com/hakimel/reveal.js/issues/745) with various work-arounds. In particular, the `r-stack` class seems to be advocated. I think it should be added as example to https://revealjs.com/fragments/.

```html
<div class="r-stack">
	<p class="fragment fade-in-then-out">They said monday.</p>
	<p class="fragment fade-in-then-out">They said mundane.</p>
</div>
```
To come up directly with the first fragment rather than having one extra step to fade it in:
```html

<div class="r-stack">
	<p class="fragment fade-out" data-fragment-index="0">They said monday.</p>
	<p class="fragment fade-in-then-out" data-fragment-index="0">They said mundane.</p>
</div>
```

**Comment by hakimel** [comment]
I agree this should be mentioned in the fragment docs. If anyone wants to add it the page is here https://github.com/reveal/revealjs.com/blob/master/src/fragments.md
